src/Bicep.Decompiler/BicepHelpers/SyntaxHelpers.cs

```diff
@@ -124,6 +124,7 @@ public static string EscapeBicepString(string value)
             "equals" => CreateToken(TokenType.Equals, "=="),
             "and" => CreateToken(TokenType.LogicalAnd, "&&"),
             "or" => CreateToken(TokenType.LogicalOr, "||"),
+            "coalesce" => CreateToken(TokenType.DoubleQuestion, "??"),

```

Comment:
Please add a test to Bicep.Decompiler.IntegrationTests to assert this is working

Reply:
Decompiler for this case (arg >2) will not work. It also does not work for `and(variable('x1'), variable('x2'), variable('x3'))` due to constraint on binary functions:
 https://github.com/Azure/bicep/blob/fe902b6cdd49931c4b5d478b1c3fecc2efedd9d7/src/Bicep.Decompiler/TemplateConverter.cs#L189

As  @majastrz [suggested](https://github.com/Azure/bicep/issues/613#issuecomment-769380888) for now we will not handle this case and we'll leave it for implementation in #1387

Reply:
That was for code gen. @anthony-c-martin does the decompiler handle `and()` or `or()` with more than 2 arguments today?

---
In reply to: [573174091](https://github.com/Azure/bicep/pull/1489#discussion_r573174091) [](ancestors = 573174091)

Reply:
Nope:
```
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "resources": [],
  "outputs": {
    "andExampleOutput": {
      "type": "bool",
      "value": "[and(bool('true'), bool('false'), bool('false'))]"
    }
  }
}
```
`Decompilation failed with fatal error "Expected 2 parameters for binary function and"`

Reply:
I don't think I was aware that `and` & `or` supported >2 params when I implemented that! I'm fine with either leaving this as a limitation (please create an issue to track if so), or fixing it in this PR - up to you.

Reply:
I'll leave it as it is now, it seems more complex since I'd need to create separate operator for it or redesign the approach. as I wrote - let's improve this in #1387